data_steward/retraction/retract_deactivated_pids.py

```diff
@@ -0,0 +1,497 @@
+# Python imports
+import re
+import argparse
+import logging
+import os
+from datetime import datetime
+
+# Third party imports
+import pandas as pd
+from jinja2 import Environment
+
+# Project imports
+from utils import bq
+from retraction.retract_utils import DEID_REGEX
+from constants.cdr_cleaner import clean_cdr as clean_consts
+from sandbox import check_and_create_sandbox_dataset
+from constants import bq_utils as bq_consts
+
+LOGGER = logging.getLogger(__name__)
+LOGS_PATH = 'LOGS'
+
+jinja_env = Environment(
+    # help protect against cross-site scripting vulnerabilities
+    autoescape=True,
+    # block tags on their own lines
+    # will not cause extra white space
+    trim_blocks=True,
+    lstrip_blocks=True,
+    # syntax highlighting should be better
+    # with these comment delimiters
+    comment_start_string='--',
+    comment_end_string=' --')
+
+TABLE_INFORMATION_SCHEMA = jinja_env.from_string("""
+SELECT *
+FROM `{{project}}.{{dataset}}.INFORMATION_SCHEMA.COLUMNS`
+""")
+DEACTIVATED_PIDS_QUERY = jinja_env.from_string("""
+SELECT DISTINCT *
+FROM `{{project}}.{{dataset}}.{{table}}`
+""")
+RESEARCH_ID_QUERY = jinja_env.from_string("""
+SELECT DISTINCT research_id
+FROM `{{project}}.{{prefix_regex}}_combined._deid_map`
+WHERE person_id = {{pid}}
+""")
+CHECK_PID_EXIST_QUERY = jinja_env.from_string("""
+SELECT
+COUNT(*) AS count
```

Comment:
I saw Krishna use `COALESCE(COUNT(*), 0) AS count` and asked him about it.  This will ensure a value is always returned, either 0 or a positive number.  Where `COUNT(*) AS count` could return an empty value.  Then you can do away with line 269:  `return bool(check_pid_df.get_value(0, 'count') > 0)` and replace it with `return check_pid_df.get_value(0, 'count')`.  zero is falsy in python and positive numbers are truthy.

Reply:
Awesome, thank you!